packages/mystmd-py/src/mystmd_py/main.py

```diff
@@ -4,41 +4,87 @@
 import sys
 import re
+import platformdirs
 
 
-def main():
-    NODE_LOCATION = (
-        shutil.which("node") or shutil.which("node.exe") or shutil.which("node.cmd")
+NODEENV_VERSION = "18.0.0"
+
+
+def find_installed_node():
+    return shutil.which("node") or shutil.which("node.exe") or shutil.which("node.cmd")
+
+
+def find_nodeenv_path():
+    return platformdirs.user_data_path(
+        appname="myst", appauthor=False, version=NODEENV_VERSION
     )
-    PATH_TO_BIN_JS = (pathlib.Path(__file__).parent / "myst.cjs").resolve()
 
-    if not NODE_LOCATION:
+
+def create_nodeenv(env_path):
+    command = [
+        sys.executable,
+        "-m",
+        "nodeenv",
+        "-v",
+        f"--node={NODEENV_VERSION}",
+        "--prebuilt",
+        "--clean-src",
+        env_path,
+    ]
+    subprocess.run(command, check=True)
+
+
+def find_any_node(binary_path):
+    node_path = find_installed_node()
+    if node_path is not None:
+        return pathlib.Path(node_path).absolute(), binary_path
+
+    nodeenv_path = find_nodeenv_path()
+    if not nodeenv_path.exists():
+        print(f"🔍 Couldn't find installed `node`.\n\n⚙️ Installing Node.js in {nodeenv_path}")
+        create_nodeenv(nodeenv_path)
+
+    new_path = os.pathsep.join(
+        [*binary_path.split(os.pathsep), str(nodeenv_path / "bin")]
+    )
+    return nodeenv_path / "bin" / "node", new_path
+
+
+def main():
+    # Find NodeJS (and potential new PATH)
+    binary_path = os.environ.get("PATH", os.defpath)
+    try:
+        node_path, os_path = find_any_node(binary_path)
+    except Exception as err:
         raise SystemExit(
             "You must install node >=18 to run MyST\n\n"
+            "As NodeJS could not be found, an attempt to install a local node environment was made, but failed\n\n"
             "We recommend installing the latest LTS release, using your preferred package manager\n"
             "or following instructions here: https://nodejs.org/en/download"
-        )
-    node = pathlib.Path(NODE_LOCATION).absolute()
+        ) from err
 
+    # Check version
     _version = subprocess.run(
-        [node, "-v"], capture_output=True, check=True, text=True
+        [node_path, "-v"], capture_output=True, check=True, text=True
     ).stdout
     major_version_match = re.match(r"v(\d+).*", _version)
 
     if major_version_match is None:
         raise SystemExit(f"MyST could not determine the version of Node.js: {_version}")
-
     major_version = int(major_version_match[1])
     if not (major_version in {18, 20, 22} or major_version > 22):
         raise SystemExit(
             f"MyST requires node 18, 20, or 22+; you are running node {major_version}.\n\n"
             "Please update to the latest LTS release, using your preferred package manager\n"
             "or following instructions here: https://nodejs.org/en/download"
         )
+
+    # Start MyST CLI
+    PATH_TO_BIN_JS = (pathlib.Path(__file__).parent / "myst.cjs").resolve()
     os.execve(
```

Comment:
Somewhat unrelated: There is a bug on windows (maybe others) where this is execution does not get keyboard commands (so the y/n input stuff doesn't work at all). Do you have any idea how to fix that @agoose77?

Reply:
I can't immediately see why the exec isn't working properly. will need to play around on Windows later.